plugins/jspsych-audio-keyboard-response.js

```diff
@@ -72,12 +79,9 @@ jsPsych.plugins["audio-keyboard-response"] = (function() {
     }
 
     // set up end event if trial needs it
-
     if(trial.trial_ends_after_audio){
       if(context !== null){
-        source.onended = function() {
-          end_trial();
-        }
+        source.addEventListener('ended', end_trial);
       } else {
```

Comment:
Oh wait I see the difference, the event listener is called on different objects. Beyond the scope of this PR, but if these ever get refactored, it would be better to separate the `audio`- and `source`-related operations to their own functions to make the logic easier to follow.

Reply:
Thanks for taking a look at this @chrisbrickhouse! Yes there's a lot of refactoring that could be done, including separating the audio and source related operations as you suggest. I had to stop myself from doing too much refactoring and standardizing across plugin files here because the goal is just get 6.2 out soon. But I think it's worth coming back to this in the next release so that the plugin files are as easy to understand as possible.

Reply:
p.s. please let us know if you notice any other issues that should be fixed before this is merged.

Reply:
Just chiming in that I agree. This might fit under #341 and other general efforts to standardize chunks of code that get used in multiple plugins.